classlib/src/main/java/org/teavm/classlib/java/lang/TDouble.java

```diff
@@ -201,7 +202,9 @@ public boolean equals(Object other) {
         if (this == other) {
             return true;
         }
-        return other instanceof TDouble && doubleToLongBits(((TDouble) other).value) == doubleToLongBits(value);
+        return other instanceof TDouble && PlatformDetector.isJavaScript()
```

Comment:
This is not correct condition. You must check if other is TDouble, and then perform platform type check. What actually happens: if you have other is Double *and* platform is JavaScript, use JavaScript-specific comparison, otherwise perform IEEE754 comparison. So actually passing something else than TDouble would result in CCE.